src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -643,6 +699,47 @@ void TriggeredPublisher::Configure(const Entity &,
 
   this->workerThread =
       std::thread(std::bind(&TriggeredPublisher::DoWork, this));
+  this->serviceWorkerThread =
+      std::thread(std::bind(&TriggeredPublisher::DoServiceWork, this));
+}
+//////////////////////////////////////////////////
+
+#define HANDLE_REQUEST(requestT, typeString) \
+if (((serviceInfo.reqType == typeString) | serviceInfo.reqType.empty()) \
+  && !isProcessing) \
+{ \
+  this->HandleRequest<requestT>(serviceInfo); \
+  isProcessing = true; \
+} \
```

Comment:
Let's not use macros for this; they don't do type checking and can be dangerous.  In this case, I think that you should be able to get away with a templated function, something like:

```
template<typename T>
handle_request(const ServiceOutputInfo & info, const std::string & typestring)
{
    if ((info.reqType == typestring) || (info.reqType.empty()) && !isProcessing) {
      this->HandleRequest<T>(info);
      isProcessing = true;
    }
}
```